recipes-devtools/valgrind/patches/0001-xen-begin-syswrap-additions-post-4.5.patch

```diff
@@ -0,0 +1,933 @@
+From: Eric Chanudet <chanudete@ainfosec.com>
+Date: Fri, 15 Feb 2019 01:46:47 -0500
+Subject: [PATCH 1/2] xen: begin syswrap additions post 4.5.
+
+Add interfaces versions for domctl/sysctl up to 4.10.
+
+Syswrap implemented:
+- XENVER_build_id
+- xsm_op FLASK_CONTEXT_TO_SID
+- domctl set_gnttab_limits
+- XEN_SYSCTL_topologyinfo/XEN_SYSCTL_cputopoinfo
+- XEN_SYSCTL_numainfo
+- get/settscinfo for post 4.9
+- XEN_SYSCTL_cpu_featureset
+- XEN_DOMCTL_set_target
+
+To be continued...
+
+xl list and xl create (with a stubdomain) should warn only on OpenXT
+specific (in this case V4V ioctl) when creating a guest with cdrom
+and/or disk and stubdomain, with a flask policy.
+
+Signed-off-by: Eric Chanudet <chanudete@ainfosec.com>
+---
+ coregrind/m_syswrap/syswrap-xen.c | 287 ++++++++++++++++++++++++++----
+ include/vki/vki-xen-domctl.h      |  65 ++++++-
+ include/vki/vki-xen-evtchn.h      |   7 +
+ include/vki/vki-xen-gnttab.h      |   1 +
+ include/vki/vki-xen-memory.h      |   1 +
+ include/vki/vki-xen-schedop.h     |   2 +
+ include/vki/vki-xen-sysctl.h      |  84 ++++++++-
+ include/vki/vki-xen-version.h     |   7 +
+ include/vki/vki-xen-x86.h         |   5 +
+ include/vki/vki-xen-xsm.h         |   2 +
+ 10 files changed, 419 insertions(+), 42 deletions(-)
+
+diff --git a/coregrind/m_syswrap/syswrap-xen.c b/coregrind/m_syswrap/syswrap-xen.c
+index 20341c4a6..1ac71fe39 100644
+--- a/coregrind/m_syswrap/syswrap-xen.c
++++ b/coregrind/m_syswrap/syswrap-xen.c
+@@ -375,6 +375,12 @@ PRE(xsm_op)
+                 sizeof(op->u._union._field))
+ 
+    switch (op->cmd) {
++   case VKI_FLASK_CONTEXT_TO_SID:
++      PRE_XEN_XSM_OP_READ(CONTEXT_TO_SID, sid_context, size);
++      PRE_MEM_READ("FLASK_CONTEXT_TO_SID",
++                   (Addr)op->u.sid_context.context.p,
++                   op->u.sid_context.size * sizeof (char));
++      break;
+    case VKI_FLASK_SID_TO_CONTEXT:
+       PRE_XEN_XSM_OP_READ(SID_TO_CONTEXT, sid_context, sid);
+       PRE_XEN_XSM_OP_READ(SID_TO_CONTEXT, sid_context, size);
+@@ -535,6 +541,13 @@ PRE(xen_version)
+       /* No inputs */
+       break;
+ 
++   case VKI_XENVER_build_id: {
++      vki_xen_build_id_t *bi = (vki_xen_build_id_t*)ARG2;
++      PRE_MEM_READ("VKI_XENVER_build_id len",
++		   (Addr)&bi->len, sizeof(vki_uint32_t));
++      break;
++   }
++
+    default:
+       bad_subop(tid, layout, arrghs, status, flags,
+                 "__HYPERVISOR_xen_version", ARG1);
+@@ -584,6 +597,11 @@ PRE(sysctl) {
+    case 0x00000009:
+    case 0x0000000a:
+    case 0x0000000b:
++   case 0x0000000c:
++   case 0x0000000d:
++   case 0x0000000e:
++   case 0x0000000f:
++   case 0x00000010:
+ 	   break;
+    default:
+       bad_intf_version(tid, layout, arrghs, status, flags,
+@@ -626,6 +644,11 @@ PRE(sysctl) {
+ 	 break;
+       case 0x0000000a:
+       case 0x0000000b:
++      case 0x0000000c:
++      case 0x0000000d:
++      case 0x0000000e:
++      case 0x0000000f:
++      case 0x000000010:
```

Comment:
typo extra 0?

Reply:
Urg, correct I did not made the patch from the proper branch it seems.